packages/vscode-extension/src/builders/customBuild.ts

```diff
@@ -0,0 +1,262 @@
+import path from "path";
+import fs from "fs";
+import os from "os";
+import fetch from "node-fetch";
+import { mkdtemp, readdir } from "fs/promises";
+import { finished } from "stream/promises";
+
+import { Logger } from "../Logger";
+import { command, exec, lineReader } from "../utilities/subprocess";
+import { CancelToken } from "./cancelToken";
+import { DevicePlatform } from "../common/DeviceManager";
+import { getAppRootFolder } from "../utilities/extensionContext";
+
+type UnixTimestamp = number;
+
+type EasBuildUrl = {
+  platform: DevicePlatform;
+  binaryUrl: string;
+  completedAt: UnixTimestamp;
+};
+
+type IsoTimestamp = string; // e.g. "2024-09-04T09:44:07.001Z"
+type UUID = string;
+type Version = string; // e.g. "50.0.0"
+
+type EASBuild = {
+  id: string;
+  status: "FINISHED" | "CANCELLED" | string;
+  platform: "ANDROID" | "IOS";
+  artifacts: {
+    buildUrl: string;
+    applicationArchiveUrl: string;
+  };
+  initiatingActor: {
+    id: UUID;
+    displayName: string;
+  };
+  project: {
+    id: UUID;
+    name: string;
+    slug: string;
+    ownerAccount: {
+      id: UUID;
+      name: string;
+    };
+  };
+  distribution: string; // e.g. "INTERNAL";
+  buildProfile: string; // e.g. "development"
+  sdkVersion: Version;
+  appVersion: Version;
+  appBuildVersion: string;
+  gitCommitHash: string;
+  gitCommitMessage: string;
+  priority: string; // e.g. "NORMAL_PLUS"
+  createdAt: IsoTimestamp;
+  updatedAt: IsoTimestamp;
+  completedAt: IsoTimestamp;
+  expirationDate: IsoTimestamp;
+  isForIosSimulator: false;
+};
+
+export async function runExternalBuild(
+  cancelToken: CancelToken,
+  platform: DevicePlatform,
+  externalCommand: string
+): Promise<string> {
+  const { stdout, lastLine: binaryPath } = await runExternalScript(cancelToken, externalCommand);
+
+  let easBinaryPath = await downloadAppFromEas(stdout, platform, cancelToken);
```

Comment:
I understand this code is generic and not specific to EAS, why we use EAS in the method name then?

Reply:
It is specific to EAS, we assume certain JSON structure.